app/modules/template/service.server.ts

```diff
@@ -0,0 +1,240 @@
+import {
+  AssetStatus,
+  type Organization,
+  type Template,
+  type User,
+} from "@prisma/client";
+import { db } from "~/database";
+import { getPublicFileURL, parseFileFormData } from "~/utils/storage.server";
+import { createNote } from "../asset";
+
+export async function createTemplate({
+  name,
+  type,
+  description,
+  signatureRequired,
+  userId,
+  organizationId,
+}: Pick<Template, "name" | "type" | "description" | "signatureRequired"> & {
+  userId: User["id"];
+  organizationId: Organization["id"];
+}) {
+  // Count the number of templates of same type for the user
+  const sameExistingTemplateCount = await db.template.count({
+    where: { type, userId },
+  });
+
+  const data = {
+    name,
+    type,
+    description,
+    signatureRequired,
+    creator: {
+      connect: {
+        id: userId,
+      },
+    },
+    organization: {
+      connect: {
+        id: organizationId,
+      },
+    },
+    isDefault: sameExistingTemplateCount === 0,
+  };
+
+  const template = await db.template.create({ data });
+
+  return template;
+}
+
+export async function updateTemplate({
+  id,
+  name,
+  description,
+  signatureRequired,
+  userId,
+}: Pick<Template, "id" | "name" | "description" | "signatureRequired"> & {
+  userId: User["id"];
+}) {
+  const data = {
+    name,
+    description,
+    signatureRequired,
+  };
+
+  const updatedTemplate = await db.template.update({
+    where: { id },
+    data,
+  });
+
+  /**
+   * If the signatureRequired is true, we need to search through all the Custodies that
+   * have this tempalate associated with it. We will check if the templateSigned is false.
+   *
+   * If it is false, this could mean a scenario that the custodian has the asset in custody
+   * and wasn't required to sign the template. But since we are setting signatureRequired to true,
+   * we need to set the asset custory to "AVAILABLE" and furthermore, ask the custodian to sign
+   * the template via mailing them.
+   */
+  if (signatureRequired === true) {
+    const custodies = await db.custody.findMany({
+      where: {
+        templateId: id,
+        templateSigned: false,
+      },
+      include: {
+        custodian: true,
+      },
+    });
+
+    for (const custody of custodies) {
+      // Set the asset status to AVAILABLE
+      await db.asset.update({
+        where: {
+          id: custody.assetId,
+        },
+        data: {
+          status: AssetStatus.AVAILABLE,
+        },
+      });
+
+      // Send notifications
+      await createNote({
+        content: `The PDF template **${updatedTemplate.name}** now requires a signature. **${custody.custodian.name}** needs to sign the **${updatedTemplate.name}** template before receiving custody.`,
+        type: "UPDATE",
+        userId,
+        assetId: custody.assetId,
+      });
+    }
+  }
+
+  return updateTemplate;
+}
+
+export async function updateTemplatePDF({
```

Comment:
@jurrejansen @rajdip-b @carlosvirreira this is the same case. We have to consider weather we want to allow this. I think once a template has been attached to even 1 custody, we should not allow changing of the PDF.

Reply:
I agree. This will reduce a lot of problems. Plus, me and @carlosvirreira also decided on (i think) to not let the users update the template PDF or any other settings once it is associated to a custody (ASSIGNED)

Reply:
Agreed. Let's implement it like that. Once a template has been used even once, we don't allow it to be edited anymore. They have to make a new one.